stroom-search/stroom-search-elastic/src/main/java/stroom/search/elastic/indexing/ElasticIndexingFilter.java

```diff
@@ -639,9 +611,46 @@ private void indexDocuments() {
         }
     }
 
+    /**
+     * Elasticsearch rejected the indexing request, because it is overloaded and
+     * cannot queue the batched payload. Retry after a delay, if we haven't exceeded the retry
+     * limit. Otherwise, terminate this thread and create an `Error` stream.
+     * @param e
+     */
+    private void handleElasticsearchException(final RuntimeException e) {
+        String statusName = "";
+        if (e instanceof ElasticsearchStatusException) {
+            statusName = ((ElasticsearchStatusException) e).status().name();
+        }
+        final String errorDetailMsg = e.getMessage() != null ? e.getMessage().substring(0,
+                Math.min(ES_MAX_EXCEPTION_CHARS, e.getMessage().length())) : "";
+        if (e instanceof ElasticsearchOverloadedException || ES_TOO_MANY_REQUESTS_STATUS.equals(statusName)) {
+            final ElasticIndexingConfig indexingConfig = elasticConfigProvider.get().getIndexingConfig();
+            if (currentRetry < indexingConfig.getRetryCount()) {
+                final long sleepDurationMs = indexingConfig.getInitialRetryBackoffPeriodMs() *
+                        ((long) currentRetry + 1);
+                try {
+                    LOGGER.warn("Indexing request by pipeline '" + pipelineName + "' was rejected by " +
+                            "Elasticsearch. Retrying in " + sleepDurationMs + " milliseconds " +
+                            "(retries: " + currentRetry + ")");
+                    Thread.sleep(sleepDurationMs);
+                } catch (InterruptedException ex) {
+                    fatalError("Indexing terminated after " + currentRetry + " retries: " +
```

Comment:
I realise this has been merged already, but can you restore the interrupt flag before throwing, i.e. 
```
Thread.currentThread().interrupt();
```